modules/21_tax/on/equations.gms

```diff
@@ -292,11 +292,53 @@ q21_taxrevFlex(t,regi)$( t.val ge max(2010, cm_startyear) ) ..
 q21_taxrevImport(t,regi,tradePe)..
   v21_taxrevImport(t,regi,tradePe)
   =e=
-*** import tax level * world market bioenergy price * bioenergy import
-  p21_tau_Import(t,regi,tradePe) * pm_pvp(t,tradePe) / pm_pvp(t,"good") * vm_Mport(t,regi,tradePe)
-    - p21_taxrevImport0(t,regi,tradePe)
+***---------------------------------------------------------------------------
+*'  import taxation: 1. "worldPricemarkup" = import tax level * world market price * tradePE import
+*'                   2. "CO2taxmarkup" = import tax level * national carbon price * imported carbon by carrier
+*'                   3. "avCO2taxmarkup" = import tax level * max( national carbon price, average carbonprice) * imported carbon by carrier
+* NOTE: In case of "CO2taxmarkup" and "avCO2taxmarkup" there is double-taxation of the CO2-content of the imported energy carrier: Once when being imported (at the border) and once when being converted to Secondary Energy (normal CO2price applied by REMIND)
+***---------------------------------------------------------------------------
+sum(tax_import_type_21, p21_tau_Import(t, regi, tradePe, tax_import_type_21)$sameas(tax_import_type_21, "worldPricemarkup")
```

Comment:
I am confused by the splitting of this code. 

If I understand it correctly, the equation has three different right sides, depending on whether tax_import_type_21 is a, b or c. 
I would find it clearer to write each of the options individually and put one "$$sameas(tax_import_type_21,a)" behind it, but not have the $ command twice in each version of the right side: 
so instead of 
`p21_tau_Import(t, regi, tradePe, tax_import_type_21)$sameas(tax_import_type_21, "CO2taxmarkup") * pm_taxCO2eqSum(t,regi) * pm_cintraw(tradePe) * vm_Mport(t,regi,tradePe) - p21_taxrevImport0(t,regi,tradePe,tax_import_type_21)$sameas(tax_import_type_21, "CO2taxmarkup")`
 I would write  
```+(  p21_tau_Import(t, regi, tradePe, tax_import_type_21) * pm_taxCO2eqSum(t,regi) * pm_cintraw(tradePe) * vm_Mport(t,regi,tradePe) - p21_taxrevImport0(t,regi,tradePe,tax_import_type_21)
 )$sameas(tax_import_type_21, "CO2taxmarkup")```
 
 (and the same for the other two cases. 
 For me this makes it easier to see that this is really just a way of saying "only one of these 3 versions will be active, and you don't need to look into the contents of the ones that are not active"

Reply:
Nice code cleaning, thank you! We have changed the equation according to your suggestion, see last commit!